### ISSUE TITLE
Make app installable on iOS

### ISSUE THREAD

**Issue by aristath** [xwp/pwa-wp#118]
I don't have an iOS device to test with, but I think this is also needed:

```html
<meta name="apple-mobile-web-app-capable" content="yes">
```

**Comment by aristath** [comment]
I know this looks like a scenario out a nightmare, but this is what it took to make the app installable on iOS:
```php
<meta name="apple-mobile-web-app-capable" content="yes" />
<meta name="mobile-web-app-capable" content="yes" />
<meta name="apple-touch-fullscreen" content="YES" />
<link rel="apple-touch-startup-image" href="<?php echo esc_url_raw( $icon_url ); ?>">
<link href="<?php echo esc_url_raw( $icon_url ); ?>" rel="apple-touch-icon">
<meta name="apple-mobile-web-app-title" content="<?php echo esc_attr( get_bloginfo( 'name' ) ); ?>">
<meta name='application-name' content='<?php echo esc_attr( get_bloginfo( 'name' ) ); ?>'>
```

**Comment by westonruter** [comment]
> `<link href="<?php echo esc_url_raw( $icon_url ); ?>" rel="apple-touch-icon">`

I believe `apple-touch-icon` is handled by core in `wp_site_icon()`:

```php
	$icon_180 = get_site_icon_url( 180 );
	if ( $icon_180 ) {
		$meta_tags[] = sprintf( '<link rel="apple-touch-icon-precomposed" href="%s" />', esc_url( $icon_180 ) );
	}
```

Aside, the latest beta of Safari is still not respecting the icons in the Web App Manifest: https://medium.com/dev-channel/progressive-web-app-progress-in-ios-12-2-beta-1-build-16e5181f-a18cd05ca361

> `<meta name="apple-touch-fullscreen" content="YES" />`

As for `apple-touch-fullscreen`, I think this should only be done if the web app manifest has `display` set to `fullscreen` and perhaps `standalone`.

> `<link rel="apple-touch-startup-image" href="<?php echo esc_url_raw( $icon_url ); ?>">`

Would this need to be different than the other icon?

> `<meta name="apple-mobile-web-app-title" content="<?php echo esc_attr( get_bloginfo( 'name' ) ); ?>">`
> <meta name='application-name' content='<?php echo esc_attr( get_bloginfo( 'name' ) ); ?>'>

Probably could simply be the same value as the `name` for the web app manifest, yes?

> `<meta name="apple-mobile-web-app-capable" content="yes" />`
> `<meta name="mobile-web-app-capable" content="yes" />`

These don't seem controversial.

**Comment by westonruter** [comment]
Seems like these would be good to just drop in to `manifest_link_and_meta`:

https://github.com/xwp/pwa-wp/blob/9205c4a7a2b5f62d1fdb257da500c113d2da7ecc/wp-includes/class-wp-web-app-manifest.php#L51-L64

Note that this function should do `$manifest = $this->get_manifest()` and use the values returned where possible as opposed to trying to re-compute them. In this way we can rely on the `web_app_manifest` filter being the source of truth.

**Comment by aristath** [comment]
> I believe apple-touch-icon is handled by core in wp_site_icon():

Nice, I didn't know that.

> Aside, the latest beta of Safari is still not respecting the icons in the Web App Manifest

Yeah, that's why I had to manually add them... everything seems to be a separate meta on iOS.

> As for apple-touch-fullscreen, I think this should only be done if the web app manifest has display set to fullscreen and perhaps standalone.

Of course. The one I was working one was standalone so it made sense to add it. 


> `<link rel="apple-touch-startup-image" href="<?php echo esc_url_raw( $icon_url ); ?>">`
> Would this need to be different than the other icon?
> ` <meta name="apple-mobile-web-app-title" content="<?php echo esc_attr( get_bloginfo( 'name' ) ); ?>">`

No, it was the same.

> <meta name='application-name' content=''>
> Probably could simply be the same value as the name for the web app manifest, yes?

Correct.

> These don't seem controversial.

:man_shrugging: 

> Seems like these would be good to just drop in to manifest_link_and_meta:

Yeah... I hate the idea of polluting it but I see no other option.

> Note that this function should do $manifest = $this->get_manifest() and use the values returned where possible as opposed to trying to re-compute them. In this way we can rely on the web_app_manifest filter being the source of truth.

Of course.
PR coming soon.

**Comment by westonruter** [comment]
Should be fixed by #129.
